src/com/palmergames/bukkit/towny/command/TownyAdminCommand.java

```diff
@@ -722,14 +725,36 @@ public void adminSet(String[] split) throws TownyException {
 			resident.setSurname(surname + " ");
 			TownyUniverse.getDataSource().saveResident(resident);
 
-			if (resident.hasSurname()){
+			if (resident.hasSurname()) {
 				TownyMessaging.sendMessage(sender, String.format(TownySettings.getLangString("msg_set_surname"), resident.getName(), resident.getSurname()));
 				TownyMessaging.sendMessage(resident, String.format(TownySettings.getLangString("msg_set_surname"), resident.getName(), resident.getSurname()));
 			} else {
 				TownyMessaging.sendMessage(sender, String.format(TownySettings.getLangString("msg_clear_title_surname"), "Surname", resident.getName()));
 				TownyMessaging.sendMessage(resident, String.format(TownySettings.getLangString("msg_clear_title_surname"), "Surname", resident.getName()));
 			}
 
+		} else if (split[0].equalsIgnoreCase("plot")) {
+			TownBlock tb = TownyUniverse.getTownBlock(player.getLocation());
+			if (tb != null) {
+				if (split.length < 2) {
+					sender.sendMessage(ChatTools.formatTitle("/townyadmin set plot"));
+					sender.sendMessage(ChatTools.formatCommand("Eg", "/ta set plot", "[town name]", ""));
+				}
+				try {
+					Town newTown = TownyUniverse.getDataSource().getTown(split[1]);
+					if (newTown != null) {
+						tb.setType(TownBlockType.RESIDENTIAL);

```

Comment:
I'm 99% sure that this will set the old town's perm line to the plot, this line probably needs to be done after the town is set. I haven't tested this in game yet.